docs/pages/guides/server-components.mdx

```diff
@@ -388,37 +327,112 @@ async function ExpensiveTask() {
 
 If you remove the `Suspense` wrapper around `<ExpensiveTask>`, you'll see that the `Loading...` waits for both components to finish rendering before updating the UI. This enables you to control the loading state incrementally. Sometimes, it makes sense to wait for everything to load at once (most of the time), while other times, it is beneficial to stream back UI as soon as you have (like the text response in ChatGPT).
 
-## Build-time rendering
+## Secrets
 
-Expo Router supports two different modes of rendering Server Components: build-time rendering and request-time rendering. These modes can be indicated on a per-route basis by using the `unstable_settings` export:
+Server Components can access secrets and server-side APIs. You can use the `process.env` object to access environment variables. You can ensure a module never runs on the client by importing the `server-only` module in your project.
 
-```tsx app/index.tsx
-import { Text, View } from 'react-native';
+```tsx actions/renderData.tsx
+// This will assert if the module runs on the client.
+import 'server-only';
 
-export const unstable_settings = {
-  // This component will be rendered at build-time and never re-rendered in production.
-  render: 'static',
-};
+import { Text } from 'react-native';
+
+export async function renderData() {
+  // This code only runs on the server.
+  const data = await fetch('https://my-endpoint/', {
+    headers: {
+      Authorization: `Bearer ${process.env.SECRET}`,
+    },
+  });
+
+  // ...
+  return <div />;
+}
+```
+
+You can define the secret in your **.env** file:
+
+```text .env
+SECRET=123
+```
+
+> You do not need to restart the dev server to update environment variables. They are automatically reloaded on every request.
+
+## Platform detection
+
+To detect which platform your code is bundled for, use the `process.env.EXPO_OS` environment variable. For example, `process.env.EXPO_OS === 'ios'`. Prefer this to `Platform.OS` as `react-native` is not fully optimized for React Server Components yet and won't work as expected.
 
+You can detect if code is running in the server by performing a `typeof window === 'undefined'` check. This will always return `true` on client devices and `false` on the server.
```

Comment:
```suggestion
You can detect if code is running on the server by performing a `typeof window === 'undefined'` check. This will always return `true` on client devices and `false` on the server.
```